Library/Homebrew/requirements/maximum_macos_requirement.rb

```diff
@@ -12,7 +12,11 @@ def initialize(tags)
 
   def message
     <<-EOS.undent
-      OS X #{@version.pretty_name} or older is required.
+      This formula either does not compile or function as expected on newer
```

Comment:
Yeah. I’m not a native speaker but here is how I understand the sentence:
> This formula either (does not compile) or (function as expected)

Moving the “either” sounds weird:
> This formula does not either compile or function as expected

Again I’m not a native speaker and @tdsmith might have a more informed opinion.